### ISSUE TITLE
Possible memory leak in 0.13 release on Ubuntu 18.04

### ISSUE THREAD

**Issue by VasiliiGeneralov** [build2/build2#102]
The v0.13 toolchain installation process crashes with 'Killed' or throws a "bad_alloc" exception:

"+ build2/build2/b-boot configure ... " starts to consume more and more RAM, occupies the whole RAM and swap and then crashes. The problem was observed using bash, sh, -j 1, -j 4, gcc(7, 9, 10) and clang(11) in all possible combinations.

My system:
16GB ram, 4 core intel i5-4590 @ 3.30GHz

$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

$ uname -a
Linux KW 5.4.0-47-generic #51~18.04.1-Ubuntu SMP Sat Sep 5 14:35:50 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by VasiliiGeneralov** [comment]
tee'ed output for convenience, please consider checking the attachment
[Build2ToolchainInstallation.log](https://github.com/build2/build2/files/5208698/Build2ToolchainInstallation.log)

**Comment by boris-kolpackov** [comment]
Thanks for running the test though the result is not very illuminating because the program is killed and so there is no stack to examine. Instead, we would want the allocation to fail so that `bad_alloc` gets thrown. One way to do that would be to temporarily disable memory overcommit:

```
cat /proc/sys/vm/overcommit_memory  # Remember the original value (probably 0).
echo 2 | sudo tee /proc/sys/vm/overcommit_memory >/dev/null
cat /proc/sys/vm/overcommit_memory # Confirm it is not 2.

<run the test>

echo 0 | sudo tee /proc/sys/vm/overcommit_memory >/dev/null # Restore the original value.
```

**Comment by VasiliiGeneralov** [comment]
The new log is in the attachment.
[BadAlloc.log](https://github.com/build2/build2/files/5209356/BadAlloc.log)

**Comment by boris-kolpackov** [comment]
Thanks, now we are getting somewhere! From the stack trace I can see that we are probing for `cli`. I assume you have a program by this name but it's not the one we are looking for. So we expect our probing to fail but instead we seem to be getting endless data from its `stdout`. So I am interested what happens when you run the following on your machine (which is essentially what we do in `build2`):

```
cli --build2-metadata=1 </dev/null 2>/dev/null
echo $?
```

**Comment by VasiliiGeneralov** [comment]
The error code is 130.
$ echo $?
130

The funniest thing is that, that "cli --build2-metadata=1 </dev/null 2>/dev/null" gives me output from my local cli tool, which I've been working on for some time now, and it spams to stdout without stop

**Comment by boris-kolpackov** [comment]
Ok, now it all makes sense. I've fixed this in commit 131c8cb and the change has been [staged](https://build2.org/community.xhtml#stage) in case you would like to verify it work. Thanks for helping to debug this!
